### ISSUE TITLE
LLM Integration Strategy - Choose Implementation Path

### ISSUE THREAD

**Issue by mnesler** [mnesler/attractor-tcg-solid-js#5]
# EDH Assistant LLM Integration - Implementation Options

## Context

We need to implement an LLM-powered assistant for EDH/Commander deck building. This issue tracks three different architectural approaches, each with different tradeoffs.

### Data Sources
- **HuggingFace Dataset** (`MechaCroc/magic-the-gathering`): 54.7k cards, likely 2-3 years old
- **Scryfall API**: Live data updated daily, includes latest sets (Marvel Super Heroes June 2026), EDHREC rankings, prices
- **Existing Integration**: We already have Scryfall integration at `src/lib/server/scryfall.ts`

### Model Options
- **Kimi k2/k2.5**: General LLM with strong reasoning, probably has MTG knowledge from web scraping
- **Fine-tuned MTG Models**: Specialized models like `TrevorJS/mtg-mistral-7b-instruct-sft-merged` with deep MTG knowledge
- **LoRA Adapters**: `minimaxir/magic-the-gathering-flan-t5-xl` - efficient fine-tuning approach

### [Option A: Kimi k2 + Scryfall RAG](#2)
**Recommended for: Quick MVP, minimal infrastructure**

**Architecture**: Scryfall data → Kimi k2 reasoning → Strategic advice

**Pros**:
- ✅ Fast to implement (1-2 days)
- ✅ Uses existing Scryfall integration
- ✅ Always up-to-date with latest cards
- ✅ No model hosting required
- ✅ Strong strategic reasoning

**Cons**:
- ❌ Not MTG-specialized
- ❌ API costs per query (~$0.01)

**Cost**: ~$0.01 per query  
**Timeline**: 1-2 days  
**[Full details in Issue #2](#2)**

---

### [Option B: Fine-tuned MTG Model (LoRA)](#3)
**Recommended for: Deep MTG knowledge, offline capability**

**Architecture**: User query → Fine-tuned Mistral-7B → MTG-specialized response

**Pros**:
- ✅ Deep MTG rules and card knowledge
- ✅ Understands card interactions perfectly
- ✅ Can work offline
- ✅ Fixed cost (hosting not per-query)

**Cons**:
- ❌ Requires GPU hosting (16GB+ VRAM)
- ❌ Training data cutoff (outdated on new sets)
- ❌ May hallucinate fake cards
- ❌ Setup complexity

**Cost**: $100-300/month (cloud) or $1500-3000 (local GPU)  
**Timeline**: 3-5 days (using pre-trained) or 1-2 weeks (training LoRA)  
**[Full details in Issue #3](#3)**

---

### [Option C: Hybrid (Scryfall + Kimi k2 + MTG Model)](#4) 🏆
**Recommended for: Best quality, production-ready**

**Architecture**: Multi-stage pipeline combining all approaches

```
User Query →
  ├─ Scryfall: Live card data
  ├─ Kimi k2: Strategic analysis
  └─ MTG Model: Validate interactions (optional)
  → Synthesized Response
```

**Pros**:
- ✅ Best of all worlds
- ✅ Scryfall keeps data fresh
- ✅ Kimi k2 provides reasoning
- ✅ MTG model validates complex interactions
- ✅ Adaptive (skip validation for simple queries)
- ✅ Highest accuracy

**Cons**:
- ❌ More complex architecture
- ❌ Higher latency (5-8s)
- ❌ Slightly higher cost

**Cost**: ~$0.014 per query (MVP) or ~$0.024 (with validation)  
**Timeline**: 3-7 days (MVP in 1-2 days)  
**[Full details in Issue #4](#4)**

---

## Recommendation Matrix

| Priority | Recommended Option | Why |
|----------|-------------------|-----|
| **Speed to market** | Option A | Fastest implementation |
| **Budget-conscious** | Option A | Pay-per-query, no hosting |
| **Best quality** | Option C | Combines all strengths |
| **Offline capability** | Option B | Self-hosted model |
| **Latest card data** | Option A or C | Uses Scryfall |
| **MTG accuracy** | Option C | Validation layer |

## Decision Criteria

Choose **Option A** if:
- You want to ship quickly (days not weeks)
- You don't have GPU infrastructure
- You're okay with $0.01/query costs
- You want minimal maintenance

Choose **Option B** if:
- You have GPU infrastructure or budget
- You need offline capability
- You want deep MTG rules understanding
- You're okay with data staleness

Choose **Option C** if:
- You want production-grade quality
- You can invest 1 week of development
- You're okay with complexity
- User satisfaction is top priority

## Next Steps

1. **Review** each option's detailed issue (#2, #3, #4)
2. **Decide** on approach based on priorities
3. **Assign** issue to implementer
4. **Track** progress in assigned issue
5. **Close** other options or mark as future work

## Related Resources
- Scryfall API: https://scryfall.com/docs/api
- HuggingFace MTG Models: https://huggingface.co/models?search=magic%20the%20gathering
- Existing integration: `src/lib/server/scryfall.ts`
- Attractor MCP: http://127.0.0.1:3001/mcp

## Questions?
Comment below or in the specific option issues.
